contrib/vagrant/provision-util.sh

```diff
@@ -179,6 +185,156 @@ os::provision::get-network-plugin() {
+  sed -i.bak -e 's+\(Type=\)notify+\1simple+' \
+      /lib/systemd/system/flanneld.service
+
+  # Changing the flanneld unit type prevents ExecStartPost from
+  # triggering at the correct time.  The post start task
+  # (mk-docker-opts.sh) must instead be run manually once flannel has
+  # written its configuration.
+  sed -i.bak -e 's+\(ExecStartPost=.*\)+#\1+' \
+      /lib/systemd/system/flanneld.service
+
+  systemctl enable flanneld
+  systemctl start flanneld
+
+  # Ensure that docker on the nodes is configured for flannel.
+  if [ "${is_master}" != "true" ]; then
+    os::provision::flannel-configure-docker
+  fi
+}
+
+os::provision::flannel-configure-master() {
+  local master_ip=$1
+  local origin_root=$2
+  local config_root=$3
+  local if_name=$4
+
+  local cert_path="${config_root}/openshift.local.config/master"
+  local ca_file="${cert_path}/ca.crt"
+  local cert_file="${cert_path}/master.etcd-client.crt"
+  local key_file="${cert_path}/master.etcd-client.key"
+  local etcd_url="https://${master_ip}:4001"
+  local etcd_key="/atomic.io/network"
+
+  local daemon_opts="-iface=${if_name} -ip-masq=false -v=5 \
+-etcd-cafile=${ca_file} -etcd-certfile=${cert_file} -etcd-keyfile=${key_file} \
+-listen=${master_ip}:8080"
+
+  cat <<EOF > /etc/sysconfig/flanneld
+FLANNEL_ETCD="${etcd_url}"
+FLANNEL_ETCD_KEY="${etcd_key}"
+FLANNEL_OPTIONS="${daemon_opts}"
+EOF
+
+  local etcdctl_cmd="${origin_root}/_tools/etcd/bin/etcdctl \
+-C ${etcd_url} --ca-file ${ca_file} --cert-file ${cert_file} --key-file \
+${key_file}"
+
+  local msg="etcd daemon to become available"
+  local condition="os::provision::check-etcd ${etcdctl_cmd}"
+  os::provision::wait-for-condition "${msg}" "${condition}"
+
+  cat <<EOF > /etc/flannel-config.json
+{
+"Network": "10.1.0.0/16",
+"SubnetLen": 24,
+"Backend": {
+    "Type": "udp",
+    "Port": 8285
+ }
+}
+EOF
+
+  # Import default configuration into etcd
+  ${etcdctl_cmd} set "${etcd_key}/config" < /etc/flannel-config.json \
+      > /dev/null
+}
+
+os::provision::check-etcd() {
+  $@ ls &> /dev/null
+}
+
+os::provision::flannel-configure-docker() {
+  local msg="flannel configuration to become available"
+  local condition="test -f /run/flannel/subnet.env"
+
+  # The master flannel instance may take a while to become available.
+  os::provision::wait-for-condition "${msg}" "${condition}" \
+      "${OS_WAIT_FOREVER}"
+
+  # Generate the docker configuration from the flannel configuration
+  /usr/libexec/flannel/mk-docker-opts.sh -d /run/flannel/docker
+
+  # Update the docker unit file to use the flannel configuration.
+  local docker_unit="/lib/systemd/system/docker.service"
+  sed -i.bak -e 's+\(Type=notify\)+\1\nEnvironmentFile=/run/flannel/docker+'\
+      "${docker_unit}"
+  sed -i.bak -e 's+\($DOCKER_NETWORK_OPTIONS\)+\1 $DOCKER_OPTS+' \
+      "${docker_unit}"
+
+  echo "Reloading docker"
+  systemctl daemon-reload
+  systemctl stop docker
+  iptables -t nat -F
```

Comment:
I read somewhere (maybe [1]) that it made sense to flush iptables when reconfiguring docker networking in case there were stale rules.  Do you want me to remove this line?  

1: https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/scratch.md#docker